backend/LexBoxApi/Auth/RequireAudienceAttribute.cs

```diff
@@ -0,0 +1,23 @@
+﻿using LexCore.Auth;
+
+namespace LexBoxApi.Auth;
+
+//todo for now this attribute only supports a single audience, this may be fine for now.
+//once we are at dotnet 8 we can support multiple audiences using the new IAuthorizationRequirementData api
+//https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?view=aspnetcore-7.0#iauthorizationrequirementdata
+public class RequireAudienceAttribute : LexboxAuthAttribute
+{
+    /// <param name="audience">audience allowed to access this endpoint</param>
+    /// <param name="exclusive">when false the default audience is also allowed, when true the default audience is not allowed</param>
+    public RequireAudienceAttribute(LexboxAudience audience, bool exclusive = false) : base(audience.PolicyName(exclusive))
```

Comment:
This seems like a more obvious default to me. Eg. why wouldn't we restrict the reset-password endpoint to JWT's that we generate for that purpose?
```suggestion
    public RequireAudienceAttribute(LexboxAudience audience, bool exclusive = true) : base(audience.PolicyName(exclusive))
```

Reply:
because we use the same endpoint when a user is logged in already to change their password. We could change that but that was way out of scope for this task. Basically in most cases the jwt you get from login should be able to do everything, we might introduce a super user mode (like in github when you go to change security it asks for your password again) which has more privilege but we don't need that.

Reply:
Oh , right 👍